File: 준영/프로그래머스/LV1/폰켓몬.py
def solution(nums):
    answer = 0
    n = len(nums)
    nums = list(set(nums))
    m = len(nums)

    if n // 2 < m:
        answer = n // 2
    else:
        answer = m
    
    return answer

nums = [3,1,2,3]
print(solution(nums))
nums = [3,3,3,2,2,4]
print(solution(nums))
nums = [3,3,3,2,2,2]
print(solution(nums))

File: 준영/프로그래머스/LV1/폰켓몬.js
function solution(nums) {
  var answer = 0
  const set_nums = new Set(nums)
  var n = nums.length
  var m = set_nums.size
  if (n/2 < m) {
    answer = n/2
  } else {
    answer = m
  }
  return answer
}


nums = [3, 1, 2, 3]
console.log(solution(nums))